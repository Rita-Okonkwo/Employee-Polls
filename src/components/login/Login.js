import { Body1, Button, Caption1, Dropdown, Label, Title1, Option } from "@fluentui/react-components"
import { useStyles } from "./Login.style"

const Login = () => {
    const styles = useStyles()
    const handleSubmit = (event) => {
        event.preventDefault()
        console.log('submitted')
        //TODO: redirect to home page
    }
    return (
        <div className={styles.rootContainer}>
            <Title1 className={styles.title1}>Employee Polls</Title1>
            <div className={styles.formContainers}>
                <Label id='select-user' required ><Body1>Please select a user</Body1></Label>
                <form method='post' onSubmit={handleSubmit}>
                    <div>
                        <Dropdown aria-labelledby='select-user' placeholder="-">
                            <Option><Caption1>Test</Caption1></Option>
                        </Dropdown>
                    </div>
                    <Button type='submit' className={styles.button}>Submit</Button>
                </form>
            </div>
        </div>
    )
}

export default Login