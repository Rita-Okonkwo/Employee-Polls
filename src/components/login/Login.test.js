import { render } from "@testing-library/react";
import {LoginC} from "./Login";
import { BrowserRouter } from "react-router-dom";

describe('Login component', () => {
    it('will render correctly', () => {
        const users = ['sarahedo', 'tylermcginnis']
        const view = render(<BrowserRouter><LoginC users={users}/></BrowserRouter>)
        expect(view).toMatchSnapshot()
    })
})