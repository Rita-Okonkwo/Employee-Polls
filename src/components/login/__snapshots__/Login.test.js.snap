// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component will render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
      <div
        class="___10nvq15_0000000 f1l02sjl f22iagw f1vx9l62 f122n59 f4d9j23"
      >
        <span
          class="fui-Title1 fui-Text ___1b2btr9_10j82vo fk6fouc fccw675 f1ebx5kk fl43uef fpgzoln f22iagw f6juhto f1mtd64y f1y7q3j9 f2jf649 f1wtj1j6 f61x5iw f8a2r77 fw4zf5m f1s9ku6b"
        >
          Employee Polls
        </span>
        <div
          class="___1fu4i2d_0000000 f22iagw f1vx9l62 f122n59 fdq70pc"
        >
          <label
            class="fui-Label ___1yevxz2_yljzsa0 fk6fouc f19n0e5 fkhj508 f1i3iumi"
            id="select-user"
          >
            <span
              class="fui-Body1 fui-Text ___nzdt000_uwwb5n0 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
            >
              Please select a user
            </span>
            <span
              aria-hidden="true"
              class="fui-Label__required ___1dfrsq9_q52x0p0 f1whyuy6 fycuoez"
            >
              *
            </span>
          </label>
          <form
            method="post"
          >
            <div>
              <div
                class="fui-Dropdown ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
              >
                <button
                  aria-expanded="false"
                  aria-labelledby="select-user"
                  class="fui-Dropdown__button ___19fe7ry_zvs8600 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl fxc4j92 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fkhj508 figsok6 f1i3iumi f1sbtcvk f11gcy0p fdghr9 f1e60jzv"
                  role="combobox"
                  type="button"
                >
                  -
                  <span
                    class="fui-Dropdown__expandIcon ___5y6ahy0_3cm10l0 f1ewtqcl fxkbij4 ftgm304 fe5j1ua f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
                  >
                    <svg
                      aria-hidden="true"
                      class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 20 20"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div />
    <div
      class="___10nvq15_0000000 f1l02sjl f22iagw f1vx9l62 f122n59 f4d9j23"
    >
      <span
        class="fui-Title1 fui-Text ___1b2btr9_10j82vo fk6fouc fccw675 f1ebx5kk fl43uef fpgzoln f22iagw f6juhto f1mtd64y f1y7q3j9 f2jf649 f1wtj1j6 f61x5iw f8a2r77 fw4zf5m f1s9ku6b"
      >
        Employee Polls
      </span>
      <div
        class="___1fu4i2d_0000000 f22iagw f1vx9l62 f122n59 fdq70pc"
      >
        <label
          class="fui-Label ___1yevxz2_yljzsa0 fk6fouc f19n0e5 fkhj508 f1i3iumi"
          id="select-user"
        >
          <span
            class="fui-Body1 fui-Text ___nzdt000_uwwb5n0 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
          >
            Please select a user
          </span>
          <span
            aria-hidden="true"
            class="fui-Label__required ___1dfrsq9_q52x0p0 f1whyuy6 fycuoez"
          >
            *
          </span>
        </label>
        <form
          method="post"
        >
          <div>
            <div
              class="fui-Dropdown ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
            >
              <button
                aria-expanded="false"
                aria-labelledby="select-user"
                class="fui-Dropdown__button ___19fe7ry_zvs8600 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl fxc4j92 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fkhj508 figsok6 f1i3iumi f1sbtcvk f11gcy0p fdghr9 f1e60jzv"
                role="combobox"
                type="button"
              >
                -
                <span
                  class="fui-Dropdown__expandIcon ___5y6ahy0_3cm10l0 f1ewtqcl fxkbij4 ftgm304 fe5j1ua f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
                >
                  <svg
                    aria-hidden="true"
                    class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Nav component will render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
      <div
        class="___10nvq15_0000000 f1l02sjl f22iagw f1vx9l62 f122n59 f4d9j23"
      >
        <span
          class="fui-Title1 fui-Text ___1b2btr9_10j82vo fk6fouc fccw675 f1ebx5kk fl43uef fpgzoln f22iagw f6juhto f1mtd64y f1y7q3j9 f2jf649 f1wtj1j6 f61x5iw f8a2r77 fw4zf5m f1s9ku6b"
        >
          Employee Polls
        </span>
        <div
          class="___1fu4i2d_0000000 f22iagw f1vx9l62 f122n59 fdq70pc"
        >
          <label
            class="fui-Label ___1yevxz2_yljzsa0 fk6fouc f19n0e5 fkhj508 f1i3iumi"
            id="select-user"
          >
            <span
              class="fui-Body1 fui-Text ___nzdt000_uwwb5n0 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
            >
              Please select a user
            </span>
            <span
              aria-hidden="true"
              class="fui-Label__required ___1dfrsq9_q52x0p0 f1whyuy6 fycuoez"
            >
              *
            </span>
          </label>
          <form
            method="post"
          >
            <div>
              <div
                class="fui-Dropdown ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
              >
                <button
                  aria-expanded="false"
                  aria-labelledby="select-user"
                  class="fui-Dropdown__button ___19fe7ry_zvs8600 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl fxc4j92 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fkhj508 figsok6 f1i3iumi f1sbtcvk f11gcy0p fdghr9 f1e60jzv"
                  role="combobox"
                  type="button"
                >
                  -
                  <span
                    class="fui-Dropdown__expandIcon ___5y6ahy0_3cm10l0 f1ewtqcl fxkbij4 ftgm304 fe5j1ua f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
                  >
                    <svg
                      aria-hidden="true"
                      class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 20 20"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div />
    <div
      class="___10nvq15_0000000 f1l02sjl f22iagw f1vx9l62 f122n59 f4d9j23"
    >
      <span
        class="fui-Title1 fui-Text ___1b2btr9_10j82vo fk6fouc fccw675 f1ebx5kk fl43uef fpgzoln f22iagw f6juhto f1mtd64y f1y7q3j9 f2jf649 f1wtj1j6 f61x5iw f8a2r77 fw4zf5m f1s9ku6b"
      >
        Employee Polls
      </span>
      <div
        class="___1fu4i2d_0000000 f22iagw f1vx9l62 f122n59 fdq70pc"
      >
        <label
          class="fui-Label ___1yevxz2_yljzsa0 fk6fouc f19n0e5 fkhj508 f1i3iumi"
          id="select-user"
        >
          <span
            class="fui-Body1 fui-Text ___nzdt000_uwwb5n0 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
          >
            Please select a user
          </span>
          <span
            aria-hidden="true"
            class="fui-Label__required ___1dfrsq9_q52x0p0 f1whyuy6 fycuoez"
          >
            *
          </span>
        </label>
        <form
          method="post"
        >
          <div>
            <div
              class="fui-Dropdown ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
            >
              <button
                aria-expanded="false"
                aria-labelledby="select-user"
                class="fui-Dropdown__button ___19fe7ry_zvs8600 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl fxc4j92 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fkhj508 figsok6 f1i3iumi f1sbtcvk f11gcy0p fdghr9 f1e60jzv"
                role="combobox"
                type="button"
              >
                -
                <span
                  class="fui-Dropdown__expandIcon ___5y6ahy0_3cm10l0 f1ewtqcl fxkbij4 ftgm304 fe5j1ua f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
                >
                  <svg
                    aria-hidden="true"
                    class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
